chromium_src/chrome/browser/ui/managed_ui.cc

```diff
@@ -39,16 +35,28 @@ bool ShouldDisplayManagedUi(Profile* profile) {
   }
 
   // Check if we should hide due to Brave-only management
-  policy::PolicyService* policy_service =
+  auto* profile_policy_service =
       profile->GetProfilePolicyConnector()->policy_service();
-  if (policy_service) {
-    const policy::PolicyMap& policies = policy_service->GetPolicies(
+  if (profile_policy_service) {
+    const auto& profile_policies = profile_policy_service->GetPolicies(
+        policy::PolicyNamespace(policy::POLICY_DOMAIN_CHROME, std::string()));
+
+    if (!brave_policy::HasOnlyBravePolicies(profile_policies)) {
+      return true;
+    }
+  }
+
+  // Also check browser-level policies
+  auto* browser_policy_service = g_browser_process->policy_service();
```

Comment:
I'm confused, I thought the original PR was only about hiding `your installation is managed in chrome://settings and in the hamburger menu`? Why would we need to check this if it's not checked in the original method?

Reply:
The check that exists in the repo now is only checking the profile provider (The policies associated with a profile pref). I think this code also checks the browser provider. But I think we didn't check yet if we do need to check the browser provider or not yet.

Reply:
seems like we shouldn't have to since the upstream code doesn't? Not sure why it doesn't though

Reply:
in the issue I asked for a specific example of why we'd need this